### DISCUSSION TITLE
[Feature] client side video transcoding

### DISCUSSION THREAD

**Discussion by akostadinov** [immich-app/immich#13442] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Hi,

I have a really under powered two core mini server. Mainly to save power during the 24x7 operation. It has an integrated Radeon card (GCN 3) that only supports h264 encoding via VAAPI and decoding is not supported on it in immich (although it should support hevc 8bit decoding according to wikipedia). But enabling hardware encoding does not really make a difference. I can't see any noticeable difference in encoding time. Also CPU stays 100% either way.

The machine learning tasks are slow but not that slow as the transcoding. So I was wondering whether the cli/mobile client can (optionally and maybe subject to additional setup) transcode the video to the desired format by the server upon uploading to avoid extensive wait times after upload?

Browsing photos, searching etc. works really good on that server, I don't see a reason to use a more power hungry machine. But uploading hundreds of videos takes week(s) to complete processing.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
This isn't really feasible for a handful of reasons, including that transcoding a bunch of files is probably too intensive for a phone (it'd melt through your table...) and that reimplementing all the transcoding logic on the client as well would be a major effort.

If you have another computer with some more power, you can have it chip in on the transcoding tasks: https://immich.app/docs/guides/scaling-immich

**Comment by akostadinov** [comment]
All phones should have hardware assisted transcoding so that they can record video in much higher resolution without melting. 

Also on desktop side, it may use ffmpeg that is installed by user if user so wishes. So I think it is feasible.

Some messaging platforms like telegram transcode videos on mobile client side. See https://www.reddit.com/r/Telegram/comments/8q0w7r/some_insight_to_telegram_video_encoding/

I understand this feature may not be high priority at all but I think it's valid and better stay open so some person may pick it up.
